src/Pipeline.ts

```diff
@@ -0,0 +1,108 @@
+import Commander from 'ioredis/built/utils/Commander';
+import {Command} from 'ioredis';
+import {MomentoRedisAdapter} from './momento-redis-adapter';
+import {ArgumentType} from 'ioredis/built/Command';
+import {CommandParameter} from 'ioredis/built/types';
+
+interface Pipeline {
+  length: number;
+}
+
+class Pipeline extends Commander<{type: 'pipeline'}> {
+  promise: Promise<[error: Error | null, result: unknown][] | null>;
+  resolve: (result: [error: Error | null, result: unknown][] | null) => void;
+  reject: (error: Error) => void;
+
+  private _queue: Array<Command> = [];
+  private _result: Array<unknown> = [];
+
+  momentoAdapter: MomentoRedisAdapter;
+
+  constructor(public redis: MomentoRedisAdapter) {
+    super();
+    this.momentoAdapter = redis;
+
+    this.promise = new Promise((resolve, reject) => {
+      this.resolve = resolve;
+      this.reject = reject;
+    });
+
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const _this = this;
+    Object.defineProperty(this, 'length', {
+      get: function () {
+        return _this._queue.length;
+      },
+    });
+  }
+
+  sendCommand(command: Command): unknown {
+    command.pipelineIndex = this._queue.length;
+    command.promise = this.invokeMomentoRedisClient(command.name, command.args);
+    this._queue.push(command);
+    return this;
+  }
+
+  addBatch(commands: Array<Array<unknown>>) {
+    for (let i = 0; i < commands.length; ++i) {
+      // Parse input cmd in Array format ex: ["get", "foo"]
+      const command = commands[i];
+      const commandName = command[0] as string;
+      const args = command.slice(1) as ArgumentType[];
+
+      // Invoke Command
+      const cmdToQueue = new Command(commandName, args);
+      cmdToQueue.promise = this.invokeMomentoRedisClient(
+        cmdToQueue.name,
+        cmdToQueue.args
+      );
+
+      // Push command with promise to queue
+      cmdToQueue.pipelineIndex = this._queue.length;
+      this._queue.push(cmdToQueue);
+    }
+    return this;
+  }
+
+  // Invoke function on momento adaptor
+  invokeMomentoRedisClient(
+    name: string,
+    args: CommandParameter[]
+  ): Promise<unknown> {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    if (this.momentoAdapter[name] === undefined) {
+      throw new Error(`Un-Supported Command Passed: ${name}`);
+    }
```

Comment:
If the user uses our scoped down interface, and they call a pipeline with the commands like `client.pipeline().zadd` will they get an indication the method from TS compiler that the method doesn't exist?

Reply:
No unfortunately not today.. Its because I pass back the `ChainableCommander` from IORedis [here](https://github.com/momentohq/momento-node-ioredis-client/pull/33/files#diff-62a7f791ea946a1c650fbf20fbe186c4d17a0f9dfc18a7f65f0654ac99a2f648R193)

Defined here
https://github.com/redis/ioredis/blob/af832752040e616daf51621681bcb40cab965a9b/bin/template.ts#L14-L17

I think could try to make our own version of this that uses our scoped down interface. Would need to dig in a bit more though. Was thinking maybe could do as follow up pr though/feature.